common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt

```diff
@@ -83,11 +84,10 @@ abstract class SensorReceiverBase : BroadcastReceiver() {
         }
 
         ioScope.launch {
-            val sensorDao = AppDatabase.getInstance(context).sensorDao()
-            val isChargingSensor = sensorDao.get(BatterySensorManager.isChargingState.id)
             val settingDao = AppDatabase.getInstance(context).settingsDao().get(0)
-            if (isChargingSensor != null && isChargingSensor.state == "false" &&
-                isChargingSensor.enabled && settingDao != null &&
+            val batteryStatusIntent = context.registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+            val isCharging = batteryStatusIntent?.let { BatterySensorManager.getIsCharging(it) }
+            if (isCharging == false && settingDao != null &&
```

Comment:
Suggestion: use `isCharging != true` to also skip updating if the intent returned null (unknown status)

Reply:
again a great suggestion, thank you :)